com.ca.lsp.cobol/lsp-service-cobol/src/main/java/com/ca/lsp/cobol/service/CopybookServiceImpl.java

```diff
@@ -71,28 +67,23 @@ public CopybookServiceImpl(
   }
 
   /**
-   * Store the informations about the workspace folder defined by the client IDE
+   * Store the information about the workspace folders defined by the client IDE
    *
    * @param workspaceFolders list of workspace folders sent by the client to the server
    */
   @Override
   public void setWorkspaceFolders(List<WorkspaceFolder> workspaceFolders) {
     this.workspaceFolders = workspaceFolders;
+    createPathListFromWorkspaceFolders();
+    setPathListInDependencyFile();
   }
 
-  /**
-   * @param uriForFileName of copybook found under workspace folder
-   * @return content of the file as String content
-   */
-  @Nullable
-  String retrieveContentByPath(Path uriForFileName) {
-    String content = null;
-    try (Stream<String> stream = Files.lines(uriForFileName)) {
-      content = stream.reduce((s1, s2) -> s1 + "\r\n" + s2).orElse("");
-    } catch (IOException e) {
-      log.error(e.getMessage());
-    }
-    return content;
+  private void createPathListFromWorkspaceFolders() {
+    workspaceFolderPaths = getWorkspaceFoldersAsPathList();
+  }
+
+  private void setPathListInDependencyFile() {
```

Comment:
Seems like this is the responsibility of `MyLanguageServer` on the initialization or of Guice using assisted injection.